lib/src/Dropdown/Dropdown.tsx

```diff
@@ -71,41 +71,39 @@ export const Dropdown: React.FC<DropdownProps> = React.forwardRef(
             (e: React.ChangeEvent<HTMLInputElement>) => {
                 if (props.multiple) {
-                    const current = Array.from(selectRef.current.options).find((option) => option.value == e.target.value);
+                    const current = selectRefOptions.find((option) => option.value == e.target.value);
                     current.selected = !current.selected;
                 } else {
-                    selectRef.current.value = e.target.value;
+                    selectRef.value = e.target.value;
                     setShow(false);
                 }
-                selectRef.current.dispatchEvent(new Event("change", { bubbles: true }));
-                props.multiple && onMultipleChange && onMultipleChange(getValueOfMultipleSelect(selectRef.current));
+                selectRef.dispatchEvent(new Event("change", { bubbles: true }));
+                props.multiple && onMultipleChange && onMultipleChange(getValueOfMultipleSelect(selectRefOptions));
             },
-            [isMobile, props.multiple, onMultipleChange]
+            [isMobile, props.multiple, onMultipleChange, selectRefOptions]
         );
 
         const selectAll = React.useCallback(
             (forceValue?: boolean | React.ChangeEvent<HTMLInputElement>) => {
-                Array.from(selectRef.current.options).forEach((_, i) => {
-                    const option = selectRef.current.options.item(i);
+                selectRefOptions.forEach((option: HTMLOptionElement) => {
                     if (!option.disabled) {
                         option.selected = typeof forceValue === "boolean" ? forceValue : !allSelected;
                     } else {
                         option.selected = false;
                     }
                 });
-                typeof forceValue === "boolean" && (selectRef.current.value = "");
-                selectRef.current.dispatchEvent(new Event("change", { bubbles: true }));
-                props.multiple && onMultipleChange && onMultipleChange(getValueOfMultipleSelect(selectRef.current));
+                typeof forceValue === "boolean" && (selectRef.value = "");
+                selectRef.dispatchEvent(new Event("change", { bubbles: true }));
+                props.multiple && onMultipleChange && onMultipleChange(getValueOfMultipleSelect(selectRefOptions));
             },
-            [allSelected, props.multiple]
+            [allSelected, props.multiple, selectRefOptions, selectRef]
         );
 
-        const isAllSelected = (): boolean => {
-            return Array.from(selectRef.current.options).every((_, i) => {
-                const option: HTMLOptionElement = selectRef.current.options.item(i);
+        const isAllSelected = React.useCallback((): boolean => {
+            return selectRefOptions.every((option: HTMLOptionElement) => {
                 return option.disabled ? true : option.selected;
             });
-        };
+        }, [selectRefOptions]);
```

Comment:
does all selection includes options that are disabled and not selected?

Reply:
yes, selectRefOptions returns every option within the select node